syntax = 'proto3';
package load;
import "google/protobuf/descriptor.proto";

message LoadSet{
    string file = 1;           // 文件名
    string msg = 2;            // 对应的消息结构名
}

extend google.protobuf.EnumValueOptions{
    LoadSet loadset=1000;
}

enum XmlInfo {
	Xml_none = 0;
    Xml_school = 1 [(loadset) = {file:'school.xml' msg:'SchoolConf'}];
}

message SchoolConf{
    SchoolAtr attr = 1;
    repeated SchoolSudent student = 2;       
}

message SchoolAtr{
    string name = 1;                //学校名称
    int32 student_num = 2;          //学生数量
    float grade = 3;                //学校评分
}

message SchoolSudent{
    int32 id = 1;               //学生id
    string name = 2;            //学生名称
    repeated int32 scores = 3;  //成绩集合，语文，数学，英语 （逗号分隔成绩）
}